document/core/exec/instructions.rst

```diff
@@ -2635,14 +2635,23 @@ Control Instructions
 
 5. Pop the values :math:`\val^m` from the stack.
 
-6. Let :math:`H` be the :ref:`catching exception handler <syntax-handler>` :math:`\CATCHadm\{a_x~instr_2^\ast\}^\ast\{\epsilon~\instr_3^\ast\}^?`, whose mapping from tag addresses to catch bodies corresponds to the |CATCH| and |CATCHALL| clauses, where :math:`a_x` is the tag address of each tag :math:`x`.
+6. Let :math:`F` be the :ref:`current <exec-notation-textual>` :ref:`frame <syntax-frame>`.
 
-   .. todo::
-       Step 6 needs proper writing out into a loop over x*, which looks up each corresponding a_x in the store.
+7. Let :math:`H` be the empty :ref:`catching exception handler <syntax-handler>` :math:`\CATCHadm`.
```

Comment:
`\CATCHadm` itself isn't really a legal phrase, since it merely is the header of a block-like construct.

I'd suggest something like:
```
7. For each catch clause :math:`(\CATCH~x_i~\instr_{2i}^\ast)` do:

   a. Assert: due to :ref:`validation <valid-try-catch>`, :math:`F.\AMODULE.\MITAGS[x_i]` exists.

   b. Let :math:`a_i` be the tag address :math:`F.\AMODULE.\MITAGS[x_i]`.

   c. Let :math:`H_i` be the handler :math:`\{a_i~\instr_{2i}^\ast\}`

8. Let :math:`H^\ast` be the concatenation of the handlers :math:`H_i`.

9. If there is a catch all clause :math:`(\CATCHALL~\instr_3^\ast)`, then:

    a. Let :math:`H^?` be the handler :math:`\{\epsilon~\instr_3^\ast\}`.

10. Else:

    a. Let :math:`H^?` be the empty handler :math:`\epsilon`.

11. :ref:`Enter <exec-handler-enter>` the block :math:`\val^m~\instr_1^\ast` with label :math:`L` and exception handler :math:`H`.
```
To make this work, you'll need to extend the definition of entering/exiting a handler slightly to include the label, to the form "Entering `\instr^\ast` with label L and handler H". But that would keep the abstraction level of the prose intact better.